modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/FrontendDataSet.js

```diff
@@ -230,6 +232,60 @@ const FrontendDataSet = ({
 		setItems(dataSetData.items);
 	}
 
+	useEffect(() => {
+		loadClientExtensions([
+			{
+				handler: (filterModules) => {
+					const newFilters = filterModules.map(
+						({context: filter, module}) => ({
+							...filter,
+							cxFilterImplementation: module['default'],
+						})
```

Comment:
Is the the only reason for forwarding `context` to util? I wonder if it can be avoided :thinking:  This may be related to our earlier discussion of globally set CX definitions, in `componentRoot.customRenderers.filters`. There, there is no direct context. You set a renderer name, and reference that name from the filter implementation. `loadClientExtensions` could then read `customRenderers` and resolve CX ones in the same way as in this PR, with additional wiring over renderer name. I'm not sure if that approach would be better :thinking:

Reply:
That's another valid approach that puts the burden of coallescing filters and field renderers in the server, instead of the client. Meaning that the code I'm doing to filter and reduce/map some of the fields of the FDS configuration JSON would need to be moved to the server.

What's best? :shrug: 

But since the configuring JSON is already very complex, and we don't have yet proper control over it (not typescript formal definition, and not fully clear what each field does and if some are redundant), I prefer to leave it like this for now, then maybe change it in the future once it's (at least) typed with TypeScript. To avoid breaking anything... :grimacing: